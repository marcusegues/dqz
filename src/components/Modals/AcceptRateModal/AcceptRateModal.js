// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import {
  View,
  StyleSheet,
  // $FlowFixMe
} from 'react-native';
import { AppModal } from '../AppModal';
import { RedButton } from '../../Buttons/RedButton';
import { ModalCard } from '../ModalCard';
import { CardHeader } from '../../QuestionAnswer/Cards/subcomponents/CardHeader';
import { CardHeaderSubText } from '../../QuestionAnswer/Cards/subcomponents/CardHeaderSubText';
import { moderateScale, scale, verticalScale } from '../../../styles/Scaling';
import type { TFunction } from '../../../types/generalTypes';
import { ModalCloseText } from '../ModalCloseText';

const ownStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: verticalScale(10),
  },
  textContainer: {
    flex: 1,
    flexDirection: 'column',
    paddingLeft: scale(20),
  },
  redButtonWrapper: {
    width: '95%',
    alignSelf: 'center',
  },
});

type AcceptRateModalProps = {
  modalVisible: boolean,
  setModalVisibleFalse: () => any,
  onAcceptRate: () => void,
  onDismiss: () => void,
};

type AcceptRateModalState = {
  acceptedRate: boolean,
};

class AcceptRateModalInner extends React.Component<
  AcceptRateModalProps & { t: TFunction },
  AcceptRateModalState
> {
  constructor(props) {
    super(props);
    this.state = {
      acceptedRate: false,
    };
  }

  handleAcceptRate() {
    this.setState({ acceptedRate: true }, () => this.props.onAcceptRate());
  }

  render() {
    const { setModalVisibleFalse, modalVisible, t } = this.props;
    return (
      <AppModal
        onRequestClose={setModalVisibleFalse}
        modalVisible={modalVisible}
        animationIn="slideInLeft"
        animationOut="slideOutLeft"
        onDismiss={() => {
          if (this.state.acceptedRate) {
            this.props.onDismiss();
          }
        }}
      >
        <ModalCard style={{ width: '95%' }}>
          <CardHeader text={t('acceptRateModal:modalTitle')} style={{}} />
          <CardHeaderSubText
            text={t('acceptRateModal:modalSubTitle')}
            style={{ lineHeight: moderateScale(18), marginBottom: 30 }}
          />

          <View style={ownStyles.redButtonWrapper}>
            <RedButton
              text={t('acceptRateModal:confirm')}
              onPress={() => this.handleAcceptRate()}
            />
          </View>
        </ModalCard>
        <ModalCloseText
          onModalHide={() => setModalVisibleFalse()}
          text={t('modal:closeModalText')}
        />
      </AppModal>
    );
  }
}

export const AcceptRateModal = (translate(['acceptRate'])(
  AcceptRateModalInner
): ComponentType<AcceptRateModalProps>);
