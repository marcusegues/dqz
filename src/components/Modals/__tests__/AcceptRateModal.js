import React from 'react';
import renderer from 'react-test-renderer';
import { AcceptRateModal } from '../AcceptRateModal/AcceptRateModal';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('AcceptRateModal', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<AcceptRateModal />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
