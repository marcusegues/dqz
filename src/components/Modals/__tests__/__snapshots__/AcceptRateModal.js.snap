// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptRateModal renders according to snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onDismiss={[Function]}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    onDismiss={[Function]}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 3,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "width": "85%",
          },
          Object {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          },
          Object {
            "width": "95%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "marginBottom": 31.388235294117646,
            "marginHorizontal": 31.388235294117646,
            "marginTop": 31.388235294117646,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#141414",
              "fontFamily": "roboto_regular",
              "fontSize": 24,
              "lineHeight": 62.77647058823529,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9B9B9B",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
            "lineHeight": 28.285714285714285,
            "marginBottom": 30,
            "paddingHorizontal": 31.388235294117646,
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "backgroundColor": "#DC0018",
              "borderColor": "#ddd",
              "borderRadius": 3,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 31.388235294117646,
              "width": "95%",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#DC0018",
                "borderRadius": 3,
                "flex": 1,
                "opacity": 1,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 3,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#fff",
                  "fontFamily": "roboto_medium",
                  "fontSize": 22,
                  "fontWeight": "500",
                  "paddingVertical": 31.388235294117646,
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "bottom": 50.285714285714285,
          "padding": 23.57142857142857,
          "position": "absolute",
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "roboto_regular",
            "fontSize": 22,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</Modal>
`;
